src/Behat/Config/Config.php

```diff
@@ -38,8 +51,47 @@ public function withProfile(Profile $profile): self
         return $this;
     }
 
+    public function withPreferredProfile(string $profileName): self
+    {
+        $this->settings[self::PREFERRED_PROFILE_NAME_SETTING] = $profileName;
+
+        return $this;
+    }
+
     public function toArray(): array
     {
         return $this->settings;
     }
+
+    public function toPhpExpr(): Node\Expr
+    {
+        $configObject =  $this->builderFactory->new(new FullyQualified(self::class));
```

Comment:
Each of these toPhpExpr() work in a similar way: first we use the builder factory to build an expr that builds an instance of this particular config object (an equivalent of new Config() or similar). Then we look at the settings of this object to see if can call any functions to add any setting (for example `withProfile() or similar). Finally we create the list of parameters for this object and add them, which may include any setting which has not been set via a function